src/main/java/org/opensearch/knn/indices/Model.java

```diff
@@ -56,7 +63,19 @@ public ModelMetadata getModelMetadata() {
      * @return length of model blob
      */
     public int getLength() {
-        return modelBlob.length;
+        if (getModelBlob() == null) {
+            return 0;
```

Comment:
Minor: Should this be -1 to make it clear to the caller that there is no model?

Reply:
Given that we use this in the cache, I think it should just be 0, indicating the model has a size of 0.